src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -508,7 +508,8 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 }
                 else
                 {
-                    GetEmitter()->emitIns_Mov(INS_mov, emitSize, targetReg, op1Reg, /* canSkip */ false);
+                    bool canSkip = ((targetReg == op1Reg) && (intrin.baseType == intrin.op1->gtType));
```

Comment:
This should never be `true` due to the checks above. If `targetReg == op1Reg` we would've hit the first `if` path.